### PR TITLE
Initial support for elliptical curves 

### PR DESCRIPTION
**Pull request by melowe**
1. Refactored encryption and config modules to allow support for alternative encryptors to NACL. 
2. Modified test fixture not to assume NACL. 
3. Refactored key pair loading to facilitate injection of encryptor.
4. Moved new keypair password generation out of config module as this is essentially handling cli command/actions into the config cli module. 

Cli changes
`tessera -updatepassword `requires `--encryptor.type=[NACL|EC]` option
`tessera -keygen` requires `--encryptor.type=[NACL|EC] `option or `{encryptor : {type:NACL or EC }` defining in config 
normal usage default to NACL for backward compatibility.

### CODE REVIEW

**Comment by melowe** [comment]
KeyPair config requires refactoring .

File: cli/config-cli/src/main/java/com/quorum/tessera/config/cli/DefaultCliAdapter.java [hunk 1]
```diff
@@ -138,24 +148,31 @@ public CliResult execute(String... args) throws Exception {
...
-                KeyEncryptorFactory.create(),
-                PasswordReaderFactory.create()
-            ).parse(commandLine);
+        if (commandLine.hasOption("updatepassword") && !commandLine.hasOption("encryptor.type")) {
+            throw new CliException("arg: --encryptor.type=[NACL|EC] is required for option -updatepassword");
+        }
 
-            //return early so other options don't get processed
-            return null;
+        if (!commandLine.hasOption("configfile") && !commandLine.hasOption("keygen")) {
```
Comment (namtruong) [inline cli/config-cli/src/main/java/com/quorum/tessera/config/cli/DefaultCliAdapter.java]: Need to add `&& !commandLine.hasOption("updatepassword")`

File: cli/config-cli/src/main/java/com/quorum/tessera/config/cli/DefaultCliAdapter.java [hunk 2]
```diff
@@ -138,24 +148,31 @@ public CliResult execute(String... args) throws Exception {
 
     private Config parseConfig(CommandLine commandLine) throws IOException {
 
-        if(commandLine.hasOption("updatepassword")) {
-            new KeyUpdateParser(
-                KeyEncryptorFactory.create(),
-                PasswordReaderFactory.create()
-            ).parse(commandLine);
+        if (commandLine.hasOption("updatepassword") && !commandLine.hasOption("encryptor.type")) {
+            throw new CliException("arg: --encryptor.type=[NACL|EC] is required for option -updatepassword");
```
Comment (namtruong) [inline cli/config-cli/src/main/java/com/quorum/tessera/config/cli/DefaultCliAdapter.java]: Should this also be default to NACL for backward compatibility?

File: cli/config-cli/src/main/java/com/quorum/tessera/config/cli/parsers/EncryptorConfigParser.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+        if (commandLine.hasOption("configfile")) {
+            final String path = commandLine.getOptionValue("configfile");
+            final Config config = JaxbUtil.unmarshal(filesDelegate.newInputStream(Paths.get(path)), Config.class);
+
+            if (Objects.nonNull(config.getEncryptor())) {
+                return config.getEncryptor();
+            } else if (commandLine.hasOption("keygen")) {
+                if (Objects.isNull(encryptorTypeValue)) {
+                    throw new CliException(NO_ENCRYPTOR_DEFINED_ERROR_MESSAGE);
```
Comment (namtruong) [inline cli/config-cli/src/main/java/com/quorum/tessera/config/cli/parsers/EncryptorConfigParser.java]: Can also be default to NACL I think

**Review by namtruong — APPROVED** [review]

**Pull request merged**